core/src/main/scala/org/apache/spark/deploy/ApplicationDescription.scala

```diff
@@ -41,5 +41,8 @@ private[spark] class ApplicationDescription(
     new ApplicationDescription(
       name, maxCores, memoryPerSlave, command, appUiUrl, eventLogDir, eventLogCodec)
 
+  // only valid when spark.executor.multiPerWorker is set to true
+  var maxCorePerExecutor: Option[Int] = None
```

Comment:
this is `private[spark] class` so it won't matter. Also it should be `maxCoresPerExecutor`.